cpp/tests/transform/row_bit_count_test.cu

```diff
@@ -0,0 +1,576 @@
+    {6, 7, 8},
+    {33, 34, 35, 36, 37, 38, 39},
+    {-1, -2},
+    {-10, -11, -1, -20},
+    {40, 41, 42},
+    {100, 200, 300},
+    {-100, -200, -300}};
+  // clang-format on
+
+  // floats
+  std::vector<float> ages{5, 10, 15, 20, 4, 75, 16, -16};
+  std::vector<bool> ages_validity = {1, 1, 1, 1, 0, 1, 0, 1};
+  auto ages_column =
+    cudf::test::fixed_width_column_wrapper<float>(ages.begin(), ages.end(), ages_validity.begin());
+
+  // int16 values
+  std::vector<int16_t> vals{-1, -2, -3, 1, 2, 3, 8, 9};
+  auto i16_column = cudf::test::fixed_width_column_wrapper<int16_t>(vals.begin(), vals.end());
+
+  // Assemble struct column
+  auto struct_column =
+    cudf::test::structs_column_wrapper({list, ages_column, i16_column}, struct_validity);
+
+  // wrap in a list
+  std::vector<int> outer_offsets{0, 1, 1, 3, 6, 7, 8};
+  cudf::test::fixed_width_column_wrapper<int> outer_offsets_col(outer_offsets.begin(),
+                                                                outer_offsets.end());
+  auto const size = static_cast<column_view>(outer_offsets_col).size() - 1;
+
+  cudf::test::fixed_width_column_wrapper<size_type> expected_sizes{276, 32, 520, 572, 212, 212};
+
+  return {cudf::make_lists_column(static_cast<cudf::size_type>(size),
+                                  outer_offsets_col.release(),
+                                  struct_column.release(),
+                                  cudf::UNKNOWN_NULL_COUNT,
+                                  rmm::device_buffer{}),
+          expected_sizes.release()};
+}
+
+std::unique_ptr<column> build_nested_column2(std::vector<bool> const& struct_validity)
+{
+  // List<Struct<List<List<int>>, Struct<int16>>>
+
+  // Inner list column
+  // clang-format off
+  cudf::test::lists_column_wrapper<int> list{    
+     {{1, 2, 3, 4, 5}, {2, 3}},
+     {{6, 7, 8}, {8, 9}},
+     {{1, 2}, {3, 4, 5}, {33, 34, 35, 36, 37, 38, 39}}};
+  // clang-format on
+
+  // Inner struct
+  std::vector<int16_t> vals{-1, -2, -3};
+  auto i16_column   = cudf::test::fixed_width_column_wrapper<int16_t>(vals.begin(), vals.end());
+  auto inner_struct = cudf::test::structs_column_wrapper({i16_column});
+
+  // outer struct
+  auto outer_struct = cudf::test::structs_column_wrapper({list, inner_struct}, struct_validity);
+
+  // wrap in a list
+  std::vector<int> outer_offsets{0, 1, 1, 3};
+  cudf::test::fixed_width_column_wrapper<int> outer_offsets_col(outer_offsets.begin(),
+                                                                outer_offsets.end());
+  auto const size = static_cast<column_view>(outer_offsets_col).size() - 1;
+  return make_lists_column(static_cast<cudf::size_type>(size),
+                           outer_offsets_col.release(),
+                           outer_struct.release(),
+                           cudf::UNKNOWN_NULL_COUNT,
+                           rmm::device_buffer{});
+}
+
+TEST_F(RowBitCount, NestedTypes)
+{
+  // List<Struct<List<int>, float, List<int>, int16>
+  {
```

Comment:
This still makes me wonder why not just create 3 different `TEST_F` functions instead of a single function with 3 scoped clauses.

Reply:
In this case, what would they be called, besides maybe NestedTypes1, NestedTypes2, NestedTypes3?  I like to cluster these kinds of tests together because they're all testing various combinations of the same thing (complex nested type hierarchies).

Reply:
Yeah, I think it's fine as it is. I cannot think of anything better.